### PR TITLE
fixed quality of the code

### PR DESCRIPTION
**Pull request by abdiawali**

### CODE REVIEW

**Comment by claraj** [comment]
@KhadarRashid please review

File: convert_state.py [hunk 1]
```diff
@@ -1,58 +1,64 @@
...
 
-
-import json  # import json for json processing
 def main():
     with open('data.json','r') as file:
-        data = file.read()
-        data = json.loads(data)
+        states = file.read()
+        states = json.loads(states)
+    print(states)
```
Comment (KhadarRashid) [inline convert_state.py]: Probably do not need to print the states, everyone most likely know the state names

File: convert_state.py [hunk 2]
```diff
@@ -1,58 +1,64 @@
...
-    stateAbbr = data #dictionary of state abbreviations keys and state name values 
-    stateAbbr2 = {}# dictionary of state name keys and state abbreviation values
+    stateAbbr = states    #dictionary of state abbreviations keys and state name values 
+    state = {}     # dictionary of state name keys and state abbreviation values
+    
     # state key is the key in the state dictionary. state value is the value in the dictionary 
-    for statekey, statevalue in stateAbbr.items() :
-        stateAbbr2[statevalue] =  statekey
-
+    for statekey, statevalue in stateAbbr.items():
```
Comment (KhadarRashid) [inline convert_state.py]: statekey can be made into state_key. Same with statevalue and stateAbbr

File: convert_state.py [hunk 3]
```diff
@@ -1,58 +1,64 @@
...
-        print("2. Convery abreviasion to state")
-        print(' 3. quit')
-        choice =   input("Enter choice : ")
+        print('2. Convery abbreviation to state')
+        print('3. quit')
+
+        choice = input("Enter choice : ")
 
-        if choice=="1":
+        if choice=='1':
```
Comment (KhadarRashid) [inline convert_state.py:30]: The spacing or lack of them between choice and the numbers is hard to read (Make it if choice == '1':)

File: convert_state.py [hunk 4]
```diff
@@ -1,58 +1,64 @@
...
-    else:
-        print ( "The abbreviation for " +  userInput+" is " + result)
-def convert_abbreviation_to_state(dictionary):
-    userInput   = input("Enter abbrviation name").upper()
-    result = dictionary.get(userInput)
-    if result == None:
-        print(' abreviation not found')
+
+
+def convert_state_to_abbreviation(dictionary):
```
Comment (KhadarRashid) [inline convert_state.py:43]: add docstrings under this function and all others (""" """)

File: convert_state.py [hunk 5]
```diff
@@ -1,58 +1,64 @@
...
+    user_input=input("Enter state name: ").capitalize()
+    result = dictionary.get(user_input)
+
+    if result==None:
```
Comment (KhadarRashid) [inline convert_state.py]: I personally did (is None) but besides that the lack of space between the words is hard to read

File: convert_state.py [hunk 6]
```diff
@@ -1,58 +1,64 @@
...
 
 
-
-import json  # import json for json processing
 def main():
     with open('data.json','r') as file:
-        data = file.read()
-        data = json.loads(data)
+        states = file.read()
+        states = json.loads(states)
```
Comment (KhadarRashid) [inline convert_state.py]: Maybe make states a different variable as it just overrides the previous variable

**Review by KhadarRashid — CHANGES_REQUESTED** [review]
I didn't comment on every line but the logic from one can be applied to others

**Comment by abdiawali** [comment]
@claraj  can you please review it

File: convert_state.py [hunk 7]
```diff
@@ -1,60 +1,73 @@
...
-        data = json.loads(data)
+        states = file.read()
+        abbreviation_to_name = json.loads(states)
 
-    print(data)
-    
     # Create a dictionary of states -> abbreviations, and a dictionary of abbreviations -> states 
-    stateAbbr = data #dictionary of state abbreviations keys and state name values 
-    stateAbbr2 = {}# dictionary of state name keys and state abbreviation values
+    state_abbr = abbreviation_to_name    #dictionary of state abbreviations keys and state name values
```
Comment (claraj) [inline convert_state.py]: Keeping the abbreviation_to_name dictionary is better than creating a  new variable. abbreviation_to_name is a better name than state_abbr

Also line too long PyLint check. Fixing the name and removing the comment would fix

File: convert_state.py (hunk 3 above)
Comment (claraj) [reply convert_state.py:30]: Fix the spacing

File: convert_state.py [hunk 8]
```diff
@@ -1,60 +1,73 @@
...
-        print ( "The abbreviation for " +  userInput+" is " + result)
-def convert_abbreviation_to_state(dictionary):
-    userInput   = input("Enter abbrviation name").upper()
-    result = dictionary.get(userInput)
-    if result == None:
-        print(' abreviation not found')
+
+
+def convert_state_to_abbreviation(dictionary):
+    '''
```
Comment (claraj) [inline convert_state.py]: Triple-quote docstrings so `""" converts state name to abbreviation """`

File: convert_state.py [hunk 9]
```diff
@@ -1,60 +1,73 @@
...
+    user_input=input("Enter state name: ").capitalize()
+    result = dictionary.get(user_input)
+
+    if result is None:
+        print('state not found')
+
     else:
-        print("the state with the abbreviation   " + userInput + " is " + result)
+        print ('The abbreviation for ' + user_input + ' is ' + result)
```
Comment (claraj) [inline convert_state.py]: Use a f-string

File: convert_state.py [hunk 10]
```diff
@@ -1,60 +1,73 @@
...
 
 
+def convert_abbreviation_to_state(dictionary):
+    '''
```
Comment (claraj) [inline convert_state.py]: triple quote

File: convert_state.py [hunk 11]
```diff
@@ -1,60 +1,73 @@
...
+    converts abbreviation to state name
+    '''
+    user_input = input("Enter abbreviation name: ").upper()
+    result = dictionary.get(user_input)
+
+    if result is None:
+        print('abbreviation not found')
 
+    else:
+        print('the state with the abbreviation ' + user_input + ' is ' + result)
```
Comment (claraj) [inline convert_state.py]: Use a f-string

**Review by claraj — CHANGES_REQUESTED** [review]
Looks much better! Just a few small things to fix. thanks!

**Comment by abdiawali** [comment]
@claraj made some changes can you please check.

**Pull request closed without merging**